Build system

lap00zza opened the pull request:
<!-- PULL REQUEST TEMPLATE -->
<!-- (Update "[ ]" to "[x]" to check a box) -->

**Describe this PR:**
This PR adds a NodeJS based build system to compile the SASS files to CSS and autoprefixes it automatically to support different browsers. Later this can be expanded to include tests/minification etc.

**What kind of change does this PR introduce?** (tick at least one)

- [ ] Bug fix
- [ ] Feature
- [ ] Documentation
- [ ] Refactoring
- [ ] Markup
- [ ] Style
- [x] Build
- [ ] Other (please describe):

**Other information:**
#### Setup
Prerequisite
* [NodeJS](https://nodejs.org/en/download/)

Once Node is installed, follow these steps to set up the build system
1. `npm install`
2. `gulp sass` or `./node_modules/.bin/gulp sass` depending on your OS
3. The css files will be built and copied to `assets/css` directory

#### Commands added
1. `gulp sass`
2. `gulp sass:watch`
3. `gulp minify-css`

lap00zza committed:
chore (build sass): added a gulp based build system for SASS

These are steps to set up the build system
1. `npm install`
2. `gulp sass` or `./node_modules/.bin/gulp sass` depending on your OS
3. The css files will be built and copied to `assets/css` directory

lap00zza committed:
ref (*): added newline at EOF

lap00zza committed:
chore (build sass): added auto-prefixer

lap00zza committed:
ref (*): removed license

lap00zza commented:
Did anyone test this yet?

yumiris commented:
Currently resetting my development workstation. I will let you know within a few hours how everything went.

MirisWisdom reviewed (approved):
Can confirm this works excellently!

lap00zza committed:
chore (gulp): added `sass:watch` command

This gulp command will watch the sass files for changes and compile css
automatically.

lap00zza committed:
chore (gulp): added `minify-css` command
